### ISSUE TITLE
Vipps

### ISSUE THREAD

**Issue by xGigaa** [PrivSec-dev/banking-apps-compat-report#74]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### App name

Vipps

### Link to app

https://play.google.com/store/apps/details?id=no.dnb.vipps

### App version

v2.122.0

### Country of the app

Norway

### Build Number

SP2A.220405.003.2022042000

### Device

- [ ] Pixel 6 Pro
- [ ] Pixel 6
- [ ] Pixel 5a
- [ ] Pixel 5
- [ ] Pixel 4a (5G)
- [X] Pixel 4a
- [ ] Pixel 4 XL
- [ ] Pixel 4
- [ ] Pixel 3a XL
- [ ] Pixel 3a
- [ ] Pixel 3 XL
- [ ] Pixel 3

### Google Play installed

- [X] Installed
- [ ] Not installed

### Google Play services Network permission revoked?

- [ ] Revoked
- [X] Not revoked
- [ ] I did not have Google Play services installed

### SafetyNet Enforcement

- [ ] Enforced
- [X] Not enforced
- [ ] Unsure

### Native code debugging

- [X] Enabled
- [ ] Disabled

### Stock OS compatibility

- [ ] Works
- [ ] Does not work
- [X] Not tested

### Profile app tested in

- [X] Main user profile
- [ ] Secondary user profile
- [ ] I used a work profile device manager app (Please mention more details about it down below like the app)

### Description of the app's functionality

App works as normal. I've tested sending money, paying bills and receiving money.
I haven't tested the group paying feature but i'm assuming that works as Vipps only does an initial check for google play service and root on startup.

### Are there any extra notes you think users should know about?

No workarounds performed.

**Comment by akc3n** [comment]
Thanks @xGigaa will be adding this shortly.

**Comment by okpersonvern** [comment]
Group paying feature works fine. No issues found during my usage. Same setup as OP.

**Comment by FID02** [comment]
A user in the forums posted a note on an issue with the Pixel 9 Pro Fold:

> Apparently when the Vipps app is in the Google Pixel Pro 9 Fold's inner screen, it attempts to do something graphically as a step before it proceeds to do the actual payment/send confirmation, and when it fails to do this step, it also does not send.

> The workaround for this is to simply close the fold and use the outer screen.

I don't know if this is a GrapheneOS-specific issue (I personally doubt it), but something to note in case the issue is raised here in the future.

**Comment by spring-onion** [comment]
@Am0rphous Hey, try toggling secure app spawning. That setting is only available in the owner profile and you need to reboot the device. You say you did not block the usage of play integrity, has the app used it in the past when it still worked? You could also do a test run with it blocked.

**Comment by Am0rphous** [comment]
So they basically aint gonna do anything about it. They replied:

_"Vipps performs integrity checks on both the app and your device to ensure security, in line with requirements from both banks and our own guidelines. If you are notified that your device has been modified/rooted, this means that Vipps unfortunately cannot be used on this device. Vipps uses security services such as Google Play Services on Android and corresponding integrity checks on iOS – to verify both the device and the Vipps app when it is started. We do this to meet strict security requirements from banks, so that we can offer the easiest possible payments, as well as the high security standards we ourselves set for our solutions. Unfortunately, this means that Vipps cannot be used on devices that do not pass these integrity checks."_

**Comment by spring-onion** [comment]
@Am0rphous Please tell them they can do all of that for GrapheneOS too while keeping play integrity, we have an article that goes over everything here: https://grapheneos.org/articles/attestation-compatibility-guide

There's a template you can use here: https://github.com/PrivSec-dev/banking-apps-compat-report/issues/342

**Comment by tockudex** [comment]
What about contactless payment? Touch or whatever it is called?

**Comment by Am0rphous** [comment]
Does it still work for you @Kiotrw25 ?

I get the same error. I'm gonna install Android Studio and emulate a device to use the service. I just get blocked using the app after 1 min of using it.

Edit: after using an emulated pixel 6a with Play Store through Android S. I get the same f error: "Vipps can't start. Try deleting Vipps and downloading it again from Play Store. It's also possible that your phone has been modified, If that's the case, get in touch with the store or person you bought the phone from". Got damn

**Comment by Kiotrw25** [comment]
> Does it still work for you [@Kiotrw25](https://github.com/Kiotrw25) ?
> 
> I get the same error. I'm gonna install Android Studio and emulate a device to use the service. I just get blocked using the app after 1 min of using it.
> 
> Edit: after using an emulated pixel 6a with Play Store through Android S. I get the same f error: "Vipps can't start. Try deleting Vipps and downloading it again from Play Store. It's also possible that your phone has been modified, If that's the case, get in touch with the store or person you bought the phone from". Got damn

@Am0rphous Yes, i just came to the realization that i am using newer pixel than what is listed here. Perhaps it don't really matter, Pixel 9

**Comment by Kiotrw25** [comment]
> What about contactless payment? Touch or whatever it is called?

@tockudex Never tried

**Comment by tockudex** [comment]
@Kiotrw25 I guess this requires Google Wallet or something. Perhaps someone actually made it work...

@Am0rphous for me it works using a Pixel 6 🤔

**Comment by xGigaa** [comment]
Tap to pay works pretty well for me. Vipps presents itself as a wallet app that you can select.

**Comment by tockudex** [comment]
You're still using the main profile for Vipps and Play services, right? Perhaps contactless payment breaks using a separate profile?

**Comment by Am0rphous** [comment]
That is a good clue @tockudex . I do have Vipps on a separate profile. That might be the "problem". Even though I've downloaded sandboxed GPS and given every permission available.

30 min later: I inspected the app and notet it was installed through Aurora Store. This might've impacted the app as it's outside Play Store. I enabled exploit protection compatibility mode and installed it through Play Store. **It seems to work (at least for now)**. Even after rebooting the device.

In Android Studio, I used a pixel 6A with Android 16 and Google Play Services. Installed the app on the main profile, but got the same issue. But this might be because Vipps detects it's being run in an emulator.

Anyway seems it is working for now, cheers!

**Comment by tockudex** [comment]
>installed it through Play Store

That means you used a google account, right?

Installing it on my main profile with G services and without changing the hardening settings, does not enable card tapping...

**Comment by Am0rphous** [comment]
Yes - I use a google account when using the Play Store on the separate profile.

I've set it up, but I haven't actually used it before. I can check later today.

**Comment by Am0rphous** [comment]
@tockudex I can confirm tapping with Vipps on grapheneOS works on my decice - out of the box. (It's very important it is installed from Play Store.)

@xGigaa I can confirm group paying feature fully works. Could you please update the original post and add a success mark for "Pixel 8 Pro" + a success mark on "Secondary user profile"?

**Comment by tockudex** [comment]
Since a couple of days, my Vipps suddenly stopped working on GrapheneOS.
The recent OS updates (20260711 and 20260715) [which were addressing some banking apps breaking](https://grapheneos.social/@GrapheneOS/116919768169426940), did not resolve my issue.

Upon typing my code in Vipps, I am faced with following red error message:
"Vipps can't start. Try deleting Vipps and downloading it from Play Store. It's also possible that your phone has been modified. If that's the case, get in touch with the store or person you bought the phone from."

I reinstalled Vipps using adb/Shizuku to spoof the installation source to PlayStore, but the vendor name cannot be set to com.android.vending any more, so that might be the culprit...

(Clarification: Vipps installed through Aurora Store using session installer/Shizuku)

**Comment by tockudex** [comment]
Update:
The GOS update `20260729` did not resolve the issue, even after clearing Vipps' storage and configuring it anew (compatibility mode on/off)

**Comment by bjornfor** [comment]
> I reinstalled Vipps using adb/Shizuku to spoof the installation source to PlayStore, but the vendor name cannot be set to com.android.vending any more, so that might be the culprit...

A data point: I have Vipps installed through Google Play, and it works fine.

I guess this confirms [this comment](https://github.com/PrivSec-dev/banking-apps-compat-report/issues/74#issuecomment-4231940236):

> (It's very important it is installed from Play Store.)

**Comment by tockudex** [comment]
> I guess this confirms https://github.com/PrivSec-dev/banking-apps-compat-report/issues/74#issuecomment-4231940236:
>
>   >(It's very important it is installed from Play Store.)

Yep, I guess if automatic app source spoofing (source + vendor name) was implemented as discussed in [5406](https://github.com/GrapheneOS/os-issue-tracker/issues/5406) and [5409](https://github.com/GrapheneOS/os-issue-tracker/issues/5409), Vipps (and other picky apps) would work flawlessly, even if installed from stores other than the Playstore!
